diff --git a/src/common/impl/netif_linux.c b/src/common/impl/netif_linux.c
index 67e4868..4c2be63 100644
--- a/src/common/impl/netif_linux.c
+++ b/src/common/impl/netif_linux.c
@@ -214,7 +214,7 @@ bool ffNetifGetDefaultRouteImplV4(FFNetifDefaultRouteResult* result) {
             result->preferredSourceAddrV4 = entry.prefsrc;
             if (minMetric == 0) {
                 FF_DEBUG("Found zero metric route, stopping further processing");
-                break; // Stop processing if we found a zero metric route
+                goto exit; // Stop processing if we found a zero metric route
             }
         }
     }
@@ -427,7 +427,7 @@ bool ffNetifGetDefaultRouteImplV6(FFNetifDefaultRouteResult* result) {
 
             if (minMetric == 0) {
                 FF_DEBUG("Found zero metric route, stopping further processing");
-                break; // Stop processing if we found a zero metric route
+                goto exit; // Stop processing if we found a zero metric route
             }
         }
     }
diff --git a/src/detection/gpu/gpu_linux.c b/src/detection/gpu/gpu_linux.c
index 91f6b56..e06b619 100644
--- a/src/detection/gpu/gpu_linux.c
+++ b/src/detection/gpu/gpu_linux.c
@@ -76,11 +76,16 @@ static const char* drmDetectAmdSpecific(const FFGPUOptions* options, FFGPUResult
     if (error) {
         return error;
     }
+    #if FF_HAVE_DRM || __has_include(<drm/drm.h>)
     if (ffStrbufEqualS(&gpu->driver, "radeon")) {
         return ffDrmDetectRadeon(options, gpu, buffer->chars);
     } else {
         return ffDrmDetectAmdgpu(options, gpu, buffer->chars);
     }
+    #else
+    FF_UNUSED(options, gpu)
+    return "fastfetch was built without DRM support";
+    #endif
 }
 
 static void pciDetectAmdSpecific(const FFGPUOptions* options, FFGPUResult* gpu, FFstrbuf* pciDir, FFstrbuf* buffer) {
@@ -217,12 +222,17 @@ static const char* drmDetectIntelSpecific(FFGPUResult* gpu, const char* drmKey,
         return "Failed to open drm device";
     }
 
+    #if FF_HAVE_DRM || __has_include(<drm/drm.h>)
     if (ffStrbufEqualS(&gpu->driver, "xe")) {
         return ffDrmDetectXe(gpu, fd);
     } else if (ffStrbufEqualS(&gpu->driver, "i915")) {
         return ffDrmDetectI915(gpu, fd);
     }
     return "Unknown Intel GPU driver";
+    #else
+    FF_UNUSED(gpu, fd)
+    return "fastfetch was built without DRM support";
+    #endif
 }
 
 static const char* pciDetectTempGeneral(const FFGPUOptions* options, FFGPUResult* gpu, FFstrbuf* pciDir, FFstrbuf* buffer) {
@@ -260,7 +270,12 @@ static const char* drmDetectNouveauSpecific(FFGPUResult* gpu, const char* drmKey
         return "Failed to open drm device";
     }
 
+    #if FF_HAVE_DRM || __has_include(<drm/drm.h>)
     return ffDrmDetectNouveau(gpu, fd);
+    #else
+    FF_UNUSED(gpu, fd)
+    return "fastfetch was built without DRM support";
+    #endif
 }
 
 static const char* pciDetectZxSpecific(const FFGPUOptions* options, FFGPUResult* gpu, FFstrbuf* pciDir, FFstrbuf* buffer) {
@@ -507,7 +522,11 @@ FF_A_UNUSED static const char* drmDetectAsahiSpecific(FFGPUResult* gpu, const ch
     ffStrbufAppendS(buffer, drmKey);
     FF_AUTO_CLOSE_FD int fd = open(buffer->chars, O_RDONLY | O_CLOEXEC);
     if (fd >= 0) {
+        #if FF_HAVE_DRM || __has_include(<drm/drm.h>)
         return ffDrmDetectAsahi(gpu, fd);
+        #else
+        return "fastfetch was built without DRM support";
+        #endif
     }
 
     return NULL;
diff --git a/src/logo/ascii/t.inc b/src/logo/ascii/t.inc
index 3adf478..b4a49ab 100644
--- a/src/logo/ascii/t.inc
+++ b/src/logo/ascii/t.inc
@@ -122,6 +122,18 @@ static const FFlogo T[] = {
         .colorTitle = FF_COLOR_FG_256 "32",
     },
     #endif
+    #ifdef FASTFETCH_DATATEXT_LOGO_TUNIX
+    // Tunix
+    {
+        .names = { "Tunix" },
+        .lines = FASTFETCH_DATATEXT_LOGO_TUNIX,
+        .colors = {
+            FF_COLOR_FG_CYAN,
+        },
+        .colorKeys = FF_COLOR_FG_BLUE,
+        .colorTitle = FF_COLOR_FG_CYAN,
+    },
+    #endif
     #ifdef FASTFETCH_DATATEXT_LOGO_TURKISH
     // Turkish
     {
diff --git a/src/logo/ascii/t/tunix.txt b/src/logo/ascii/t/tunix.txt
new file mode 100644
index 0000000..362162e
--- /dev/null
+++ b/src/logo/ascii/t/tunix.txt
@@ -0,0 +1,12 @@
+$1                      _.'      .
+':'.               .''   __ __  .
+  '.:._          ./  _ ''     "-'.__
+.'''-: """-._    | .                "-"._
+ '.     .    "._.'                       "
+    '.   "-.___ .        .'          .  :o'.
+      |   .----  .      .           .'     (
+       '|  ----. '   ,.._                _-'
+        .' .---  |.""  .-:;.. _____.----'
+        |   .-""""    |      '
+      .'  _'         .'    _'
+     |_.-'            '-.'
